src/compiler/checker.ts

```diff
@@ -546,7 +547,7 @@ namespace ts {
                             // - Type parameters of a function are in scope in the entire function declaration, including the parameter
                             //   list and return type. However, local types are only in scope in the function body.
                             // - parameters are only in the scope of function body
-                            if (meaning & result.flags & SymbolFlags.Type) {
+                            if (meaning & result.flags & SymbolFlags.Type && lastLocation.kind !== SyntaxKind.JSDocComment) {

```

Comment:
Can you add a brief comment as to why this check is needed here? I'm having trouble groking where this might be an issue based on how this while loop walks up the parents.